[ChocoRagdoll] iP

ChocoRagdoll opened the pull request:
### Ragdoll

> Why are ragdoll cats so cute? - Song Yichao

Ragdoll is a 🐈 chatbot that is

- **easy** to learn 
- **super easy** to use
- mostly **text-based** 
- ~~cute~~ **_extraordinarily cute_**

All you need to do is 

1. Download it from [here](https://github.com/ChocoRagdoll/ip/releases/tag/v0.1).
2. Open cmd in the folder where the Ragdoll.jar file is located.
3. Run the command `java -jar Ragdoll.jar`.
4. Add your tasks and let it manage your tasks for you.

Of course it's **FREE**!

🐈‍⬛ Features

- [x] Manage tasks
- [x] Show current date and time
- [ ] Customize username (coming soon like **legit**)
- [ ] Reminders (coming soon)
- [ ] Remastered GUI (coming soon)

---

If you are a Java programmer, you can use it as a reference to practice your coding skills too. Here are the `main` method and the `run` method:

```java
public static void main(String[] args) {
    new Ragdoll(Paths.get("data", "tasks.txt").toString()).run(); 
}

public void run() {
    ui.showWelcomeMessage();
    boolean isExit = false;

    while(!isExit) {
        String input = ui.readCommand();
        ui.showLine();

        Command c = Parser.parse(input);
        c.execute(tasks, ui, storage);
        isExit = c.isExit();

        ui.showLine();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ChocoRagdoll committed:
Changed the name to Ragdoll and
Implemented an initial skeletal version

ChocoRagdoll committed:
Echo commands and exit with 'bye'

ChocoRagdoll committed:
add task and list all function

ChocoRagdoll committed:
mark tasks as done / undone and customized personality

ChocoRagdoll committed:
added 3 task types and some implementation changes

ChocoRagdoll committed:
automated testing using i/o until 3 task types

ChocoRagdoll committed:
complementary error handling and customized msgs for exceptions

ChocoRagdoll committed:
added support for deleting and testing, fixed lowercase issue

ChocoRagdoll committed:
added application of enums and fixed some exception handling

ChocoRagdoll committed:
Match the current java coding standard

ChocoRagdoll committed:
Save the tasks on the hard disk for certain operations

ChocoRagdoll committed:
Merge branch 'branch-Level-7'

ChocoRagdoll committed:
Parse dates and add relevant functions

ChocoRagdoll committed:
Merge branch 'branch-Level-8'

ChocoRagdoll committed:
Make the code more OOP

Let's include separate classes Ui, TaskList, Storage, Parser, and all
commands that are currently in use.

This design aims to separate concerns, improve code readability, and
make the application more extensible.

ChocoRagdoll committed:
Ignore the tasks.txt file

ChocoRagdoll committed:
Merge branch 'branch-MoreOOP'

ChocoRagdoll committed:
Create packages

ChocoRagdoll committed:
Reformat the imports in TaskList

ChocoRagdoll committed:
Merge branch 'branch-A-Packages'

ChocoRagdoll committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ChocoRagdoll committed:
Add Gradle support

ChocoRagdoll committed:
Merge branch 'branch-A-Gradle'

ChocoRagdoll committed:
Add JUnit test

Let's include various tests for the Storage class and the TaskList
class so that the testing can be automated to ensure validity.

Let's test the addTask, deleteTask, listTaskOnDate in the TaskList
class, save and load in the Storage class thoroughly.

ChocoRagdoll committed:
Merge branch 'branch-A-JUnit'

ChocoRagdoll committed:
Create a .jar file

ChocoRagdoll committed:
Minor Ui fix

ChocoRagdoll committed:
Add JavaDoc comments

ChocoRagdoll committed:
Comply with the CS2103T coding standard

ChocoRagdoll committed:
Add a find command

ChocoRagdoll committed:
Merge branch 'branch-A-CodingStandard'

ChocoRagdoll committed:
Merge branch 'branch-A-JavaDoc'

ChocoRagdoll committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/core/Ui.java
#	src/main/java/tasks/TaskList.java

ChocoRagdoll committed:
Check and follow the code style

ChocoRagdoll committed:
Merge branch 'branch-A-CheckStyle'

Zer0Legion commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,19 @@
+import data.Storage;
+import tasks.TaskList;
+
+//CHECKSTYLE.OFF: MissingJavadocType
```
Should there be a descriptive header comment for the public class here?

ChocoRagdoll replied:
Yes sir absolutely! I was lazy and therefore did not want to include header for all the command classes.

Zer0Legion commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,19 @@
+public class AddCommand extends Command {
+    private String input;
+
+    public AddCommand(String input) {
```
Should there be a descriptive header comment for the public method here?

ChocoRagdoll replied:
Yes!

Zer0Legion commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,29 @@
+import data.Storage;
+import tasks.TaskList;
+
+/**
```
I like that there is a descriptive header comment for the public class, which explains the class very clearly.

ChocoRagdoll replied:
Thank you!

Zer0Legion commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,29 @@
+     * @param ui      The user interface to display messages and interact with the user.
+     * @param storage The storage to save or load data.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage);
```
I like that there is a descriptive header comment for the public method, which explains what the method does, and its parameters very clearly.

ChocoRagdoll replied:
Thanks!

Zer0Legion commented on src/main/java/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,19 @@
+package commands;
+
+import core.Ui;
```
I like how imports are arranged alphabetically from well-named packages.

ChocoRagdoll replied:
For readability!

Zer0Legion commented on src/main/java/core/Parser.java:
```diff
@@ -0,0 +1,61 @@
+package core;
+
+import commands.AddCommand;
```
I like how imports are explicit and not replaced with wildcard imports, making the functionality of the code much more readable and understandable.

ChocoRagdoll replied:
Troublesome but had to follow the standards

Zer0Legion commented on src/main/java/core/Ui.java:
```diff
@@ -0,0 +1,214 @@
+    private Scanner scanner;
+    private String user = "Master";
+    private String logo = " /$$$$$$$                            /$$           /$$ /$$\n"
+            + "| $$__  $$                          | $$          | $$| $$\n"
```
I like the wonderful text art here.

ChocoRagdoll replied:
RAGDOLL!

Zer0Legion commented on src/test/java/data/StorageTest.java:
```diff
@@ -0,0 +1,84 @@
+        assertEquals(tasksToSave.size(), loadedTasks.size(), "Loaded tasks should match saved tasks in number.");
+
+        assertTrue(loadedTasks.get(0) instanceof ToDo
+                && loadedTasks.get(0).getDescription().equals(todo.getDescription()),
```
I like how operators are placed at the start of the newline, making the code neat and readable.

ChocoRagdoll replied:
Followed the rules without any questions

Zer0Legion commented on src/test/java/tasks/TaskListTest.java:
```diff
@@ -0,0 +1,141 @@
+    public void deleteTask_wrongFormat_messagePrinted() {
+        taskList.addTask("deadline return book /by 2022-12-30", mockUi, mockStorage);
+        taskList.addTask("event project meeting /from 2022-12-30 /to 2022-12-31", mockUi, mockStorage);
+
```
I like how newlines separate logical chunks within a single block of code.

ChocoRagdoll replied:
thank you!

Zer0Legion reviewed (commented):
Generally, the code is very well structured and readable, making it a pleasant experience to review the code.

ChocoRagdoll committed:
Add a GUI to the chatbot using JavaFX

ChocoRagdoll committed:
Merge branch 'branch-Level-10'

ChocoRagdoll committed:
Use varargs

ChocoRagdoll committed:
Merge branch 'branch-A-Varargs'

ChocoRagdoll reviewed (commented):
Thanks for reviewing my code!

ChocoRagdoll committed:
Build a new Jar file

ChocoRagdoll committed:
Add in assertions

ChocoRagdoll committed:
Improve code quality

Complex boolean expressions in the codebase made it difficult to
understand the logic at a glance. Additionally, some methods were
exceeding a reasonable length, reducing readability and making
maintenance harder.

To improve code quality:
- Refactored long and complex boolean expressions into smaller, named
boolean variables that clearly describe the condition being checked.
- Broke down longer methods into smaller, more focused methods that
are easier to understand and test.

This approach enhances readability by making the logic more transparent
and expressive. It also aids future refactoring efforts by isolating
specific behaviors that can be independently modified or optimized.

No functionality has been altered; these changes are purely structural
and aim to conform to best practices in code quality.

ChocoRagdoll committed:
Merge pull request #2 from ChocoRagdoll/branch-A-Assertions

Branch-A-Assertions

ChocoRagdoll committed:
Merge branch 'master' of https://github.com/ChocoRagdoll/ip

ChocoRagdoll committed:
Merge pull request #3 from ChocoRagdoll/branch-A-CodeQuality

Improve code quality

ChocoRagdoll committed:
Merge branch 'master' of https://github.com/ChocoRagdoll/ip

ChocoRagdoll committed:
Implement Tagging functionality for Tasks

This commit introduces the ability for users to add tags to tasks in
the task management system. It allows for a more organized and
searchable task list, enhancing user experience by enabling task
categorization.

Changes include:
- Updated the Task class to include a 'tag' field along with a setter
method `setTag(String tag)` to assign tags to tasks.
- Modified the `toFileFormat` method in the Task class to include the
tag information when tasks are saved, ensuring tags are persisted
across sessions.
- Added a new method `tag(String input, Ui ui, Storage storage)`
to handle the tagging command from users. This method parses the input
to extract the task index and the desired tag, applies the tag to the
specified task, and updates the storage accordingly.
- Adjusted the Storage class's `load` method to correctly parse and
load task tags from the data file, ensuring that tags are retained
when tasks are reloaded into the system.
- Corrected and enhanced the UI feedback messages for tagging
operations to provide clear, user-friendly responses upon tagging
actions.

Note:
- Tags are prefixed with '#' in user commands but stored without the
prefix for flexibility.

This feature enriches the application's task management capabilities
by introducing a simple yet powerful way to organize and identify
tasks through tags.

ChocoRagdoll committed:
Merge branch 'branch-C-Tagging'

ChocoRagdoll committed:
Add a user guide

ChocoRagdoll committed:
Modify the readme.md file

ChocoRagdoll committed:
Delete commented-out code
